src/overlays/actors/ovl_En_Hy/z_en_hy.c

```diff
@@ -1188,46 +1431,47 @@ void EnHy_Draw(Actor* thisx, PlayState* play) {
 
     OPEN_DISPS(play->state.gfxCtx, "../z_en_hy.c", 2318);
 
-    if (this->actionFunc != EnHy_InitImpl) {
+    if (this->actionFunc != EnHy_WaitForObjects) {
         Gfx_SetupDL_25Opa(play->state.gfxCtx);
         Matrix_Translate(this->modelOffset.x, this->modelOffset.y, this->modelOffset.z, MTXMODE_APPLY);
-        envColorSeg8 = sModelInfo[this->actor.params & 0x7F].envColorSeg8;
-        envColorSeg9 = sModelInfo[this->actor.params & 0x7F].envColorSeg9;
-
-        switch (this->actor.params & 0x7F) {
-            // ENHY_TYPE_AOB
-            // ENHY_TYPE_COB
-            case ENHY_TYPE_AHG_2:
-            case ENHY_TYPE_BOJ_3:
-            case ENHY_TYPE_AHG_4:
-            case ENHY_TYPE_BOJ_5:
-            // ENHY_TYPE_BBA
-            case ENHY_TYPE_BJI_7:
-            case ENHY_TYPE_CNE_8:
-            case ENHY_TYPE_BOJ_9:
-            case ENHY_TYPE_BOJ_10:
-            case ENHY_TYPE_CNE_11:
-            case ENHY_TYPE_BOJ_12:
-            case ENHY_TYPE_AHG_13:
-            case ENHY_TYPE_BOJ_14:
-            case ENHY_TYPE_BJI_15:
-            case ENHY_TYPE_BOJ_16:
-            case ENHY_TYPE_AHG_17:
-            // ENHY_TYPE_BOB_18
-            case ENHY_TYPE_BJI_19:
-            case ENHY_TYPE_AHG_20:
+        envColorSeg8 = sModelInfo[ENHY_GET_TYPE(&this->actor)].envColorSeg8;
+        envColorSeg9 = sModelInfo[ENHY_GET_TYPE(&this->actor)].envColorSeg9;
+
+        switch (ENHY_GET_TYPE(&this->actor)) {
+            // ENHY_TYPE_DOG_LADY
+            // ENHY_TYPE_WOMAN_3
+            case ENHY_TYPE_MAN_1_BEARD:
```

Comment:
Why are some of these types commented in the switch? If their case isn't here, then the code below wouldn't run?

Reply:
Yeah the code doesn't run for those cases. The comments were just there from the original decomp I think. I should probably just remove them.

Reply:
I put those comments in, to highlight which types are *not* there (they are placed in the enum's order)
but yeah feel free to remove :shrug:

Reply:
I see, could always put the comments in the default case?

Reply:
Sure why not.